site/source/tutorials/internationalization.md

```diff
@@ -0,0 +1,504 @@
+
+In addition to the `localizeBundle` method, `@dojo/widget-core/mixins/I18n` introduces two properties: a boolean `rtl` property and a string `locale` property. The boolean `rtl` property determines whether the text decoration is right-to-left (`true`) or left-to-right(`false`: the default). The `locale` property specifies the locale used to determine which message translations are rendered. Since each widget can have its own distinct locale, it is possible to use multiple languages in the same application. If no `locale` property is included, widgets assume the root application locale (`i18n.locale`). Note, however, that since Dojo 2 widgets are controlled, child widgets do not automatically inherit the locale from their parent. Parents must pass the correct locale as needed to child widgets; otherwise, a child widget will use the application locale while its parent uses a custom locale.
+
+```typescript
+import I18nMixin, { I18nProperties } from '@dojo/widget-core/mixins/I18n';
+import greetings from './nls/greetings';
+
+export interface GreetingProperties extends I18nProperties {
+	name: string;
+}
+
+const GreetingBase = I18nMixin(WidgetBase);
+export default class Greeting extends GreetingBase<GreetingProperties> {
+	render() {
+		const messages = this.localizeBundle(greetings);
+		const { name } = this.properties;
+		const hello = messages.format('hello', { name });
+
+		return v('p', [ hello ]);
+	}
+}
+```
+
+## The Build Process
+
+Dojo 2 uses [webpack](https://webpack.js.org) to bundle applications. Since Dojo 2's packages are designed to function independently, a few extra steps are required to ensure that CLDR data and locale-specific message translations are included in the build. Dojo 2's build is configured via a `.dojorc` at the project's root directory. `.dojorc` is a JSON file containing settings defined under specific namespaces. All build options are specified under the `build-webpack` namespace, and the following are specific to internationalization:
+
+* `locale`: the default locale for the application
+* `supportedLocales`: an optional array of additional locales to include in the build
+* `messageBundles`: an optional array of any message bundles to include in the build
+
+Currently, the build does not ensure the `locale` specified in the `.dojorc` is set as the application's root locale; it is only used for bundling CLDR data and locale-specific message translations. This will change in a future release, but in the meantime, the initial locale must be set manually in the application with `@dojo/i18n/i18n.switchLocale`. Further, CLDR data are currently parsed from calls to `@dojo/i18n/cldr/load`, but this too will change in a future release to instead inject CLDR data from a `.dojorc` setting.
```

Comment:
Need to make sure that issues exist to ensure that these statements are updated when the discussed changes are made.